src/mtcp/mtcp_restart.c

```diff
@@ -652,8 +698,9 @@ unmap_memory_areas_and_restore_vdso(RestoreInfo *rinfo)
   while (mtcp_readmapsline(mapsfd, &area)) {
     if (area.addr >= rinfo->restore_addr && area.addr < rinfo->restore_end) {
       // Do not unmap this restore image.
-    } else if (mtcp_strcmp(area.name, "[vdso]") == 0) {
+    } else if (area.addr == rinfo->newVdsoStart) {
```

Comment:
No, the newVdsoStart points to the new VDSO segment we get from kernel when mtcp_restart is launched.

At some point, I intend to come back and revamp some variable names, perhaps have some sub-structs within RestoreInfo for better organization.

Reply:
We make this change here for a couple of reasons:
1. we don't have to read /proc/self/maps again to find the current vdso/vvar addresses.
2. The MANA plugin relocates vdso/vvar and given the possibility of a missing /proc/self/maps vdso/vvar label after remap, we want the plugin to directly update newVdsoStart, etc., with the relocated addresses.

Reply:
Maybe `currentVdsoStart` would be a better name than `newVdsoStart`.  I had the confusion, because I assumed that this "new" vdso was going to replace the "old" vdso.

Reply:
Fixed